### PR TITLE
Get Application Gateway Server Variables APIs changes for 2018-11-01

### PR DESCRIPTION
**Pull request by nirkum18**
### Latest improvements:
<i>This is a breaking change. Get Application Gateway Server Variables APIs changes were merged in 2018-11-01. We want the APIs results to be of type List(string) rather than Dictionay(string,List)
Link to previous PR: #4916</i>
<i>Link to the PR with changes for 2018-12-01 version: https://github.com/Azure/azure-rest-api-specs/pull/5113</i>
<i></i>

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by KrisBash — APPROVED** [review]
My understanding is that this change aligns client spec to current server behavior. Signing off for ARM

**Review by sergey-shandar — APPROVED** [review]

**Comment by sergey-shandar** [comment]
@sarangan12 could you check Ruby CI?

File: specification/network/resource-manager/readme.ruby.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ azure-arm: true
 ``` yaml $(ruby) && $(multiapi)
 batch:
   - tag: package-2018-12-only
+  - tag: package-2018-11-only
```
Comment (sarangan12) [inline specification/network/resource-manager/readme.ruby.md]: This tag is not defined in the readme.md file. Did you miss to add the changes?

**Review by sarangan12 — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/readme.ruby.md (hunk 1 above)
Comment (nirkum18) [reply specification/network/resource-manager/readme.ruby.md]: Yeah, that was missed. Added now.

**Comment by MikhailTryakhov** [comment]
@nirkum18  please apply your changes to the latest ```2018-12-01``` api version. currently PR can't be merged with changes to existing api (and not for the newest one) and will be exluded without fixes

**Comment by nirkum18** [comment]
> @nirkum18 please apply your changes to the latest `2018-12-01` api version. currently PR can't be merged with changes to existing api (and not for the newest one) and will be exluded without fixes

@MikhailTryakhov these changes have been applied to the '2018-12-01' already. But since this is a breaking change, we need to fix the code in 2018-11-01 as well.

**Comment by nirkum18** [comment]
> > @nirkum18 please apply your changes to the latest `2018-12-01` api version. currently PR can't be merged with changes to existing api (and not for the newest one) and will be exluded without fixes
> 
> @MikhailTryakhov these changes have been applied to the '2018-12-01' already. But since this is a breaking change, we need to fix the code in 2018-11-01 as well.

@MikhailTryakhov The changes present in 2018-11-01 currently are breaking and needs to be fixed. This PR contains the changes to fix them. Is there no way we can merge these changes?

**Comment by sergey-shandar** [comment]
@MikhailTryakhov, @nirkum18 so what's your decision?

**Comment by nirkum18** [comment]
> @MikhailTryakhov, @nirkum18 so what's your decision?

@sergey-shandar @MikhailTryakhov We need these changes to get in to fix the already breaking code in 2018-11-01. Else these APIs would not work for anyone using 2018-11-01.

**Comment by number213** [comment]
> @sergey-shandar @MikhailTryakhov We need these changes to get in to fix the already breaking code in 2018-11-01. Else these APIs would not work for anyone using 2018-11-01.

Next Network SDK is going to be 2018-12-01, no fixes are planned for 2018-11-01 (though I still think that fixing older specs is a good idea).

**Comment by sergey-shandar** [comment]
@nirkum18 LGTM, do you have agreement with @MikhailTryakhov and could you have a look ruby again? It looks like there is a problem with tags.

**Comment by nirkum18** [comment]
Had conversation with @MikhailTryakhov 
Since these changes will not affect the already in use real clients, getting these changes in will not bring much value.

So closing this PR.

**Pull request closed without merging**